src/test/java/com/google/api/codegen/testdata/ruby_Rakefile_library.baseline

```diff
@@ -1,3 +1,97 @@
 ============== file: Rakefile ==============
-# -*- ruby -*-
-require 'bundler/gem_tasks'
+# Copyright 2017, Google Inc. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "bundler/setup"
+require "bundler/gem_tasks"
+
+require "rubocop/rake_task"
+RuboCop::RakeTask.new
+
+desc "Run tests."
+task :test do
+  $LOAD_PATH.unshift "lib", "test"
+  Dir.glob("test/**/*_test.rb").each { |file| require_relative file }
+end
+
+namespace :test do
+  desc "Runs tests with coverage."
+  task :coverage do
+    require "simplecov"
+    SimpleCov.start do
+      command_name "library"
+      track_files "lib/**/*.rb"
+      add_filter "test/"
+    end
+
+    Rake::Task["test"].invoke
+  end
+end
+
+desc "Runs the smoke tests."
+task :smoke_test do |t, args|
+  if ENV["SMOKE_TEST_PROJECT"].nil?
+    fail "The SMOKE_TEST_PROJECT environment variable must be set. "\
+      "e.g SMOKE_TEST_PROJECT=test123 rake smoke_test"
+  end
+
+  $LOAD_PATH.unshift "lib", "smoke_test"
+  Dir.glob("smoke_test/**/*smoke_test.rb").each { |file| require_relative file }
+end
+
+namespace :smoke_test do
+  desc "Runs smoke tests with coverage."
+  task :coverage do |t, args|
+    require "simplecov"
+    SimpleCov.start do
+      command_name "library"
+      track_files "lib/**/*.rb"
+      add_filter "smoke_test/"
+    end
+
+    Rake::Task["smoke_test"].invoke
+  end
+end
+
+desc "Run the CI build"
+task :ci do
```

Comment:
I'm not seeing justifications for these ci tasks -- are there any discussions why we should have them in Rakefile?

I feel like these items will be the role of CI config files (like `.travis.yml` or `circle.yml`) -- they can specify the 'script' section; see https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/.travis.yml for example.